File: Codigos/C++/C2/Listas e Trabalhos/LISTA 28-06-18/20180628/agenda.cpp
#include "agenda.h"

agenda::agenda(){
    for(int i = 0; i < 10; i++) this->p[i] = 0;
    this->nome = "";
    this->nascimento = NULL;
    this->altura = 0;
}

void agenda::armazenarPessoa(string nome, data *nascimento, float altura){

    for(int i = 0; i < 10; i++){
        if(this->p[i]){
            pessoa* Pessoa = new pessoa(nome, nascimento, altura);
            this->p[i] = Pessoa;
            break;
        }
    }
}

void agenda ::removePessoa(string nome){

    for(int i = 0; i < 10; i++){
        if(this->nome == nome){
            this->p[i] = NULL;
        }
    }
}

int agenda::buscaPessoa(string nome){

    for(int i = 0; i < 10; i++){
        if(this->nome == nome){
            return i+1;
        }
    }
    return 0;
}

void agenda::mostraAgenda(){

    for(int i = 0; i < 10; i++){
        cout << "agenda: " << this->p[i]->getNome() << endl;
    }
}

void agenda::mostraPessoa(int index){

    if(this->p[index]){
        data* date = this->p[index]->getNascimento();
        cout << "Nome: " << this->p[index]->getNome() << endl;
        cout << "Nascimento: " << date->getDia() << "/" << date->getMes() << "/" << date->getAno() << endl;
        cout << "Altura: " << this->p[index]->getAltura() << endl;
    }
}

agenda ::~agenda(){
    cout << "Destruindo Agenda ..." << endl;
}

File: Codigos/JAVA/WorkSpace C2/Agenda/src/Model/agenda.java
package Model;

public class agenda {

	private pessoa p[];
	private int n;
	
	public agenda() {
		this.p = new pessoa[10];
		this.n = 0;
	}
	
	public void armazenaPessoa( pessoa p) {
		this.p[n++] = p;
	}
	
	public void imprimePessoa(int index) {
		System.out.println(p[index]);
	}
	
	public void imprimeAgenda() {
		System.out.println("Relação de pessoas da agenda ....");
		for(int i = 0; i < n; i++) {
			imprimePessoa(i);
		}
	}
	
	public int buscaPessoa(String nome) {
		for(int i = 0; i < n; i++) {
			if(p[i].equals(nome)) {
				return i;
			}
		}
		return -1;
	}
	
	public void removePessoa(String nome) {
		int index = buscaPessoa(nome);
		if(index >= 0) {
			for(int i = index +1; i < n; i++) {
				p[i-1] = p[i];
			}
			n--;
			p[n] = null;
		}
	}
	
	public String toString() {
		return "Agenda contendo " + this.n + " pessoas";
	}
}
